edx/commerce-coordinator
# feat: Add OrderFulfill endpoint and LMS enrollment signal

[REV-2847](https://2u-internal.atlassian.net/browse/REV-2847).

Hackathon work for the Theseus Project POC for enrollment codes - fulfill, which is separated into 3 parts:
1) Adding a `/order/fulfill` endpoint that Titan can make a `POST` request to for order fulfillment after the purchase is approved, plus a Celery task for calling LMS enrollment.
2) `POST` request to LMS enrollment to enroll the user.
3) Coupon invalidation to mark it as used.

<img width="1188" alt="Screen Shot 2022-08-28 at 11 27 41 AM" src="https://user-images.githubusercontent.com/13632680/187081909-b2e55e76-81aa-4e09-a1e0-4648bcffb54f.png">

**This PR is part 1:**
- Adds `OrderFulfill` API to Titan app, including a serializer file (`POST`)
- Adds Titan -> LMS signal and Celery task for order fulfillment

**Testing instructions:**

1. Checkout this branch `julianajlk/REV-2847/fulfill-titan`
2. Follow instructions on the README of this repo to start redis in `devstack` and start Celery in `commerce-coordinator` 
3. Go to http://localhost:8140/titan/fulfill/
4. Insert data
4. You should get a 200 OK POST response (response should be "" for now, error: false) and on the Celery task side, you should get a log with 
`Task commerce_coordinator.apps.lms.tasks.fulfill_order_placed_send_enroll_in_course_task [13604e04-1f01-4ebd-8ba1-f1188e891dbc] succeeded`
